manager-bundle/src/HttpKernel/JwtManager.php

```diff
@@ -26,30 +26,13 @@ class JwtManager
     public const COOKIE_NAME = 'contao_settings';
 
     /**
-     * @var Signer
+     * @var Configuration
      */
-    private $signer;
+    private $config;
 
-    /**
-     * @var Builder
-     */
-    private $builder;
-
-    /**
-     * @var Parser
-     */
-    private $parser;
-
-    /**
-     * @var string
-     */
-    private $secret;
-
-    public function __construct(string $projectDir, Signer $signer = null, Builder $builder = null, Parser $parser = null, Filesystem $filesystem = null)
+    public function __construct(string $projectDir, Configuration $config = null, Filesystem $filesystem = null)
```

Comment:
@leofeyer this is a BC break, right?

Reply:
Nvm, got it in 25a0b71.

Reply:
It is a BC break, however, it is also a BC break to add another argument to the constructor IMHO, because it will result in a "declaration should be compatible with parent" error. We should probably mark the `JwtManager` class as `@internal`, because people are supposed to use the service and not the class itself.

@aschempp @ausi @Toflar WDYT?

Reply:
`JwtManager` is not a service, it is injected by the kernel. But I would mark the class as `@internal` so we're free to modify that. The class itself certainly isn't API to me.

Reply:
The same question applies to the [`ContaoLoginAuthenticationListener`](https://github.com/bytehead/contao/blob/fix/2fa-php8/core-bundle/src/Security/Authentication/ContaoLoginAuthenticationListener.php).

Reply:
@aschempp The `JwtManager` is registered as a service (and public): https://github.com/contao/contao/blob/4.9/manager-bundle/src/Resources/config/services.yml#L25, but directly injected by the kernel: https://github.com/contao/contao/blob/4.9/manager-bundle/src/HttpKernel/ContaoKernel.php#L263

Reply:
> because it will result in a "declaration should be compatible with parent" error

This is not true for `__construct()` AFAIK https://3v4l.org/kFcii

Reply:
Except if the constructor is part of the interface (see https://3v4l.org/mQRa8), which is not the case here I think, is it?

Reply:
> which is not the case here I think, is it?

Correct.